osu.Game/Overlays/Direct/DirectPanel.cs

```diff
@@ -108,6 +134,15 @@ private void load(BeatmapManager beatmaps, OsuColour colours, BeatmapSetOverlay
             beatmaps.BeatmapDownloadBegan += attachDownload;
         }
 
+        protected override void LoadAsyncComplete()
+        {
+            base.LoadAsyncComplete();
+
+            // value may have been set before the panel had fully finished loading
+            if (!downloadIndicatorsVisible)

```

Comment:
we don't want to do this kind of thing. make an updateThings() method and call it in LoadComplete (and bar running it in the `set` of property using `if (IsLoaded)`. search IsLoaded across the codebase for examples.